src/exosite.c

```diff
@@ -546,7 +546,7 @@ static void exo_process_active_ops(exo_op *op, uint8_t count)
       case EXO_REQUEST_SUB_ACK_NEW:
       case EXO_REQUEST_SUB_ACK:
         // send ack for observe notification
-        exo_build_msg_ack(&pdu, coap_get_mid(&pdu));
+        exo_build_msg_ack(&pdu, op[i].mid);
```

Comment:
The function works just fine, but the in this part of the code I'm building a new pdu so the memory that this is reading is actually uninitialized garbage. I guessing that I just usually get lucky before and it was using the same bit of memory for this pdu's buffer as in the last function that actually received the original packet from the server.